Src/AmrCore/AMReX_FluxRegister.cpp

```diff
@@ -616,22 +625,40 @@ FluxRegister::ClearInternalBorders (const Geometry& geom)
             for (FabSetIter fsi(frhi); fsi.isValid(); ++fsi) {
                 const Box& bx = fsi.validbox();
                 const std::vector< std::pair<int,Box> >& isects = balo.intersections(bx);
+                auto const& frarr = frhi[fsi].array();
                 for (int ii = 0; ii < static_cast<int>(isects.size()); ++ii) {
-                    frhi[fsi].setVal<RunOn::Host>(0.0, isects[ii].second, 0, nc);
+                    if (Gpu::inLaunchRegion()) {
+                        tags.emplace_back(Array4BoxTag<Real>{frarr, isects[ii].second});
+                    } else {
+                        frhi[fsi].setVal<RunOn::Host>(0.0, isects[ii].second, 0, nc);
+                    }
                 }
                 if (geom.isPeriodic(dir)) {
                     if (bx.bigEnd(dir) == domain.bigEnd(dir)) {
                         const Box& sbx = amrex::shift(bx, dir, -domain.length(dir));
                         const std::vector<std::pair<int,Box> >& isects2 = balo.intersections(sbx);
                         for (int ii = 0; ii < static_cast<int>(isects2.size()); ++ii) {
                             const Box& bx2 = amrex::shift(isects2[ii].second, dir, domain.length(dir));
-                            frhi[fsi].setVal<RunOn::Host>(0.0, bx2, 0, nc);
+                            if (Gpu::inLaunchRegion()) {
+                                tags.emplace_back(Array4BoxTag<Real>{frarr, bx2});
+                            } else {
+                                frhi[fsi].setVal<RunOn::Host>(0.0, bx2, 0, nc);
+                            }
                         }
                     }
                 }
             }
         }
     }
+
+#ifdef AMREX_USE_GPU
+    // There is Gpu::synchronize in Parallefor below internally.
```

Comment:
Typo? `Parallefor` → `ParallelFor`
```diff
-Parallefor
+ParallelFor
```

Reply:
Yes that's a typo.

Reply:
I can open a PR to fix it, if you'd like.

Reply:
Sure.  Thanks!

Reply:
@WeiqunZhang do you mind if I fix all of the typos across the code base (using [codespell](https://github.com/codespell-project/codespell))? Things like "virtural", "retruned", "apoints", "funciton", "sequencially", "untill", "occurence", etc.

Reply:
Fine with me.

Reply:
Okay, I'll do that.  To give you a hint about the volume, there's about ~437 :smile:

Still ok with you?

Reply:
yes